wagtail/query.py

```diff
@@ -596,28 +597,53 @@ def _get_chunks(self, queryset) -> Iterable[Tuple[Dict[str, Any]]]:
         else:
             # Iterate through the queryset, returning the rows in manageable
             # chunks for self.__iter__() to fetch full instances for
-            current_chunk = []
-            for r in queryset.iterator(self.chunk_size):
-                current_chunk.append(r)
-                if len(current_chunk) == self.chunk_size:
-                    yield tuple(current_chunk)
-                    current_chunk.clear()
-            # Return any left-overs
-            if current_chunk:
-                yield tuple(current_chunk)
-
-
-class DeferredSpecificIterable(ModelIterable):
+            yield from batched(queryset.iterator(self.chunk_size), self.chunk_size)
+
+
+class DeferredSpecificIterable(SpecificIterable):
+    """
+    A highly-modified version of `ModelIterable`, which returns instances
+    of the specific model, but with only the parent model's fields.
+    """
+
     def __iter__(self):
-        for obj in super().__iter__():
-            if obj.specific_class:
-                yield obj.specific_deferred
-            else:
-                warnings.warn(
-                    "A specific version of the following object could not be returned "
-                    "because the specific model is not present on the active "
-                    f"branch: <{obj.__class__.__name__} id='{obj.id}' title='{obj.title}' "
-                    f"type='{obj.content_type}'>",
-                    category=RuntimeWarning,
-                )
-                yield obj
+        queryset = self.queryset
+        model = queryset.model
+        db = queryset.db
+        specific_models = {}
+
+        annotations = list(queryset.query.annotation_select.keys())
+
+        results = queryset.values()
```

Comment:
@RealOrangeOne My suggestion is to change this line to:
```suggestion
        results = queryset.values_list(named=True)
```

Then you use attribute-style access for the `content_type_id` and eventually `row._fields` in place of `row.keys()` and `(val for val in row)` in place of `row.values()`

Reply:
Having ported and benchmarked, the difference is fairly small, but using `.values` seems to be faster than `named=True`, ~2.4ms vs ~2.3ms. I suspect the reason is that `dict` is incredibly well optimised, since it's used basically everywhere internally in Python.

I think given that, at least in this pass, I'm going to stick to `.values`.